unknown_type/[M]230.cpp

```diff
@@ -0,0 +1,33 @@
+// Problem: 230. Kth Smallest Element in a BST
+// Link: https://leetcode.com/problems/kth-smallest-element-in-a-bst/
+
+/* Nhận xét:
+- Với một BST thì ta duyệt theo inorder traversal sẽ được một dãy tăng dần.
+- Mỗi khi duyệt đến một đỉnh, ta giảm k đi 1. Khi nào giảm tới 0 thì trả về kết quả.
+- Time complexity: O(max(log(n), k)), do phải mất log(n) bước để tìm phần tử nhỏ nhất, sau đó tìm k đỉnh tiếp theo. Trường hợp suy biến là O(n) khi BST bị suy biến.
+- Space complexity: O(1).
+*/
+
+class Solution {
+public:
+    void traverse(TreeNode *u, int &k, int &res) {
+        if (u) {
```

Comment:
Chú có thể code if (!u) { return }; cho code nó đẹp. Follow rule "return as soon as possible".

Reply:
Okay, got it!
Just a questions: Anh cũng thích kiểu code return ngay nếu được. 
Tuy nhiên trong dự án của anh, bọn nó hay dùng mấy tool static code analysis và có 1 MINOR rule của tool MISRA đó là với 1 hàm thì chỉ return tại 1 điểm do khi đọc code sẽ confused là hàm sẽ dừng ở đâu.
Chú thấy cái đấy có quan trọng không?

Reply:
Theo coding convention chuẩn thì nên return as soon as possible. A nghĩ cái đó nên follow. Nếu ko nhiều trường hợp sẽ phải if else rất phức tạp.